alamenai/rehook
# Fix: unshown confirmation message

## Description

When I implemented this hook at the first time, I extracted the confirmation message in independent function, however this led to bug where the alert message did not display at all.

## What's Changed

- [x] Refactored the `handleBeforeUnload` function.


## Review Instructions

1. Copy and paste this code on your `App.tsx` file :

```jsx
import { useUnsavedFormChanges } from './hooks/use-unsaved-form-changes'
import { useState, SyntheticEvent } from 'react'

function App() {
    const [formData, setFormData] = useState('') // Assuming your form data is stored in state
    const { setFormChanged } = useUnsavedFormChanges()

    const handleChange = (event: SyntheticEvent<HTMLInputElement>) => {
        // Update form data and mark as changed
        setFormData(event.currentTarget.value)
        setFormChanged(true)
    }

    return (
        <div style={{ display: 'flex', flexDirection: 'column', marginLeft: 900, gap: 10, fontSize: 24 }}>
            <input
                type="text"
                name="example"
                value={formData}
                onChange={handleChange}
                style={{ height: 44, width: 280, fontSize: 18, paddingInline: 12 }}
            />
            <button>Save</button>
        </div>
    )
}

export default App
```

2. Fill the input and refresh the page.
3. I expect to see an alert message to notify you that there is unsaved changes.
